src/test/kotlin/org/rust/lang/core/completion/RsDeriveCompletionProviderTest.kt

```diff
@@ -39,6 +43,15 @@ class RsDeriveCompletionProviderTest : RsCompletionTestBase() {
         struct Foo;
     """)
 
+    fun `test complete with manually implemented dependencies`() = checkSingleCompletion("Copy", """
+        #[derive(Cop/*caret*/)]
+        enum Foo { Something }
+
+        impl Clone for Foo {
+            fn clone(&self) -> Foo { Foo::Something }
+        }
+    """)
```

Comment:
Let's use `doSingleCompletion` for new completion tests. It allows to verify caret placement as well, and is much less magical then `checkSingleCompletion`: https://github.com/intellij-rust/intellij-rust/blob/a900832640fefaacf1c698b5bac69479b0791567/src/test/kotlin/org/rust/lang/core/completion/RsCompletionTest.kt#L17-L31